### ISSUE TITLE
Migrate from 0.3.X

### ISSUE THREAD

**Issue by cristianocca** [marcgibbons/django-rest-swagger#684]
Hello,

I'm trying to upgrade from 0.3.x, I know everything changed and I'm trying to see how to make it work, but I'm having all these issues:

- I have individual apps creating their own api urls, there's no django rest framework api at all since DefaultRouter is not used (causes issues), version 0.3 would still work with this and render all the urls created from SingleRouter, but it doesn't seem the case with version 2, only 1 of the N apps are actually rendering their URLs.

- Previously documentation was written through yaml comments on the ViewSet method, and it would render inputs automatically from the serializer. Right now it doesn't seem to be recognizing the serializer values at all.

Is there any guide on how to migrate?

Thanks.

**Comment by cristianocca** [comment]
After some work, made it work (barely).

Points to consider:
- Make sure all api views defines get_serializer_class, if you were like me using plain ViewSets, the api won't render any request body, you need to mix in something that defines a get_serializer method and returns the actual seriazer.
- Make sure there's a default authentication class set up, since endpoints that require authentication won't be picked up even if they define an authentication class. It's probably going to cause issues when the authentication class of the view doesn't match the default one...
- Properly set up swagger SECURITY_DEFINITIONS settings

So far this new version seems to only be a headache, and it's missing functionality, at least when using the auto docs, such as:
- Properly rendering ChoiceField, no dropdowns at all
- Not being able to render File Fields
- Not possible to define custom query string parameters
- url path params name doesn't match the ones defined on the method (ie: id instead of pk)
- Can't define custom response messages, nor response body format
- Can't use a different "parameter content type" field? Even if form and multipart parsers are defined

Version 0.3.X could still work with django 1.11 with just updating from RequestContext objects to dicts when rendering templates.

**Comment by tuffnatty** [comment]
@cristianocca Exactly. I would only recommend you to apply all my PRs or check out [my branch](https://github.com/tuffnatty/django-rest-swagger/tree/my-0.3.x) which has many more fixes other than updating from `RequestContext` objects to `dict`s when rendering templates. Also, if you really want to get OpenAPI-2.0 compliant API specifications, [here is a hack](https://gist.github.com/tuffnatty/e2090c6734355e6473316e6e3b4aac40). Other than that, try [drf-openapi](https://github.com/limdauto/drf_openapi) which seems to be moving in the right direction.

**Comment by gise88** [comment]
Hi @tuffnatty! I would like to try your branch. Do you have some documentation or better a working example? Thank you for your work!

**Comment by tuffnatty** [comment]
@gise88 just checkout the branch. It contains only fixes, no new features are added. So the documentation is here: http://django-rest-swagger.readthedocs.io/en/stable-0.3.x/
